cocotb/regression.py

```diff
@@ -525,8 +526,8 @@ def _create_test(function, name, documentation, mod, *args, **kwargs):
     Returns:
         Decorated test function
     """
-    def _my_test(dut):
-        yield function(dut, *args, **kwargs)
+    async def _my_test(dut):
+        await types.coroutine(function)(dut, *args, **kwargs)
```

Comment:
This seems pointless due to line 588, where we already know that `function` is a `cocotb.coroutine` and will therefore produce `RunningTask` instances which are already awaitable.

Reply:
It might even do the wrong thing, will change.